src/include/rbd/librbd.h

```diff
@@ -1089,9 +1089,16 @@ CEPH_RBD_API int rbd_namespace_remove(rados_ioctx_t io,
                                       const char *namespace_name);
 CEPH_RBD_API int rbd_namespace_list(rados_ioctx_t io, char *namespace_names,
                                     size_t *size);
-CEPH_RBD_API int rbd_namespace_exists(rados_ioctx_t io, const char *namespace_name,
+CEPH_RBD_API int rbd_namespace_exists(rados_ioctx_t io,
+                                      const char *namespace_name,
                                       bool *exists);
 
+CEPH_RBD_API int rbd_pool_init(rados_ioctx_t io, bool force);
+CEPH_RBD_API int rbd_pool_stats_get(rados_ioctx_t io, bool scan_snaps,
+                                    uint32_t *image_count,
+                                    uint32_t *snapshot_count,
+                                    uint64_t *provisioned_bytes);
```

Comment:
@dillaman Just wondering, won't we want some additional stats in future, like  `used_bytes`, and will need to introduce `rbd_pool_stats_get2` or something else?

I guess we could do similar to what we do with `rbd_image_options_t`. And then `rbd_pool_stats_get` would only return stats for items that were specified. Or does it look like over engineering for this task?

Reply:
I don't think it would ever be feasible to compute the used-stats online. The fastest method right now would be the ```rbd disk-usage``` approach when fast-diff is enabled. If we didn't have to worry about namespaces, we could get the used stats from the monitor.

I did just realize that I would need to handle the case for images in the trash, though.

Reply:
The used-stats was just an example, I can think out other things, like number of namespaces, number of images in the trash. Though I am almost sure the first users request when they see `rbd pool stats` will be to add the "used" here :-)

Reply:
Perhaps rename the method to something like ```rbd_pool_images_stats_get``` or  ```rbd_provisioned_image_stats_get```?

Reply:
I like ```rbd_provisioned_image_stats_get``` the most, but anyway it was just a question, and I am fine with the current version too.